src/TestFramework/Coverage/XmlReport/FileCodeCoverageProvider.php

```diff
@@ -35,43 +35,16 @@
 
 namespace Infection\TestFramework\Coverage\XmlReport;
 
-use array_key_exists;
-use Infection\TestFramework\Coverage\CoverageFileData;
-use Symfony\Component\Finder\SplFileInfo;
+use Infection\TestFramework\Coverage\CoveredFileData;
 
 /**
  * @internal
  * @final
  */
 class FileCodeCoverageProvider
 {
-    /**
-     * @var array<string, CoverageFileData>|null
-     */
-    private $coverage;
-
-    private $coverageFactory;
-
-    public function __construct(PhpUnitXmlCoverageFactory $coverageFactory)
-    {
-        $this->coverageFactory = $coverageFactory;
-    }
-
-    public function provideFor(SplFileInfo $fileInfo): FileCodeCoverage
+    public function provideFor(CoveredFileData $fileData): FileCodeCoverage
```

Comment:
Is that provider still really needed?

Reply:
Yes and no. The whole purpose of this file is to save my time on conflict resolution. I guess I can add it to some post-merge clean up list, how do you think?